geo-traits/src/geometry_collection.rs

```diff
@@ -0,0 +1,78 @@
+use crate::{Dimension, GeometryCollectionIterator, GeometryTrait};
+use geo_types::{CoordNum, Geometry, GeometryCollection};
+
+/// A trait for accessing data from a generic GeometryCollection.
+pub trait GeometryCollectionTrait: Sized {
+    /// The coordinate type of this geometry
+    type T: CoordNum;
+
+    /// The type of each underlying geometry, which implements [GeometryTrait]
+    type ItemType<'a>: 'a + GeometryTrait<T = Self::T>
+    where
+        Self: 'a;
+
+    /// The dimension of this geometry
+    fn dim(&self) -> Dimension;
+
+    /// An iterator over the geometries in this GeometryCollection
+    fn geometries(&self) -> GeometryCollectionIterator<'_, Self::T, Self::ItemType<'_>, Self> {
+        GeometryCollectionIterator::new(self, 0, self.num_geometries())
+    }
+
+    /// The number of geometries in this GeometryCollection
+    fn num_geometries(&self) -> usize;
```

Comment:
I'm more of an `n()` fella myself, but this is much more explicit. I would be just fine with this. Though an `n()` alias would also be nice :)

Reply:
I feel like I don't commonly see `n` as a shorthand for "number of things" in rust libraries. Can you point to any examples?

Even `num` is a little out there AFAIK - usually it's `len` in rust. But at least `num` is long-form enough to be clear even if it's not idiomatic.

(@JosiahParry perfect time for us to pull out that "bikeshedding" definition 😆)

Reply:
I think of `len` as "how large is this collection", but it should be 100% obvious what "this collection" means. Maybe `len` is descriptive enough for `GeometryCollection`, but it isn't for `Polygon`, because does `len()` mean the total number of rings or just the number of interiors? For that reason I like having the trait methods be very direct and descriptive, even if slightly more verbose